nodejs/src/test-helper.mjs

```diff
@@ -224,6 +232,78 @@ export default function testSuite(
 		});
 	});
 
+	describe("stream error handling", () => {
+		beforeEach(() => {
+			globalThis.fetch = vi.fn();
+		});
+
+		afterEach(() => {
+			vi.restoreAllMocks();
```

Comment:
I don't think this will remove the manually overridden property on `globalThis`.
You need to store it in a variable and reset


```
let originalFetch: typeof fetch;
beforeEach(() => {
  originalFetch = globalThis.fetch;
	globalThis.fetch = vi.fn();
});

afterEach(() => {
  globalThis.fetch = originalFetch;
});
```

Reply:
I still don't like that we're manually resetting function, instead of mocking it with vi tools. I've improved it, now it resets as expected
<img width="394" alt="Screenshot 2024-12-04 at 2 41 33 PM" src="https://github.com/user-attachments/assets/2d2a473a-da3f-4c3f-8245-d321c36a44ba">
<img width="493" alt="Screenshot 2024-12-04 at 2 42 42 PM" src="https://github.com/user-attachments/assets/50e71655-5b83-445e-8a4f-709a16d49d4e">